tests/unit/test_row_filters.py

```diff
@@ -823,83 +823,83 @@ def test_value_regex_filter___repr__():
-    from google.cloud.bigtable.row_filters import ExactValueFilter
+    from google.cloud.bigtable.row_filters import LiteralValueFilter
     from google.cloud.bigtable_v2.types import data as data_v2_pb2
 
-    value = regex = b"value-regex"
-    row_filter = ExactValueFilter(value)
+    value = regex = b"value_regex"
+    row_filter = LiteralValueFilter(value)
     expected_dict = {"value_regex_filter": regex}
     assert row_filter.to_dict() == expected_dict
     expected_pb_value = row_filter._to_pb()
     assert data_v2_pb2.RowFilter(**expected_dict) == expected_pb_value
 
 
 def test_exact_value_filter_to_pb_w_str():
-    from google.cloud.bigtable.row_filters import ExactValueFilter
+    from google.cloud.bigtable.row_filters import LiteralValueFilter
 
-    value = "value-regex"
+    value = "value_regex"
     regex = value.encode("ascii")
-    row_filter = ExactValueFilter(value)
+    row_filter = LiteralValueFilter(value)
     pb_val = row_filter._to_pb()
     expected_pb = _RowFilterPB(value_regex_filter=regex)
     assert pb_val == expected_pb
 
 
 def test_exact_value_filter_to_dict_w_str():
-    from google.cloud.bigtable.row_filters import ExactValueFilter
+    from google.cloud.bigtable.row_filters import LiteralValueFilter
     from google.cloud.bigtable_v2.types import data as data_v2_pb2
 
-    value = "value-regex"
+    value = "value_regex"
     regex = value.encode("ascii")
-    row_filter = ExactValueFilter(value)
+    row_filter = LiteralValueFilter(value)
     expected_dict = {"value_regex_filter": regex}
     assert row_filter.to_dict() == expected_dict
     expected_pb_value = row_filter._to_pb()
     assert data_v2_pb2.RowFilter(**expected_dict) == expected_pb_value
 
 
 def test_exact_value_filter_to_pb_w_int():
-    import struct
-    from google.cloud.bigtable.row_filters import ExactValueFilter
+    from google.cloud.bigtable.row_filters import LiteralValueFilter
 
     value = 1
-    regex = struct.Struct(">q").pack(value)
-    row_filter = ExactValueFilter(value)
+    # encode numbers with double escapes for the backslashes
+    expected_byte_string = b"\\\x00\\\x00\\\x00\\\x00\\\x00\\\x00\\\x00\\\x01"
+    row_filter = LiteralValueFilter(value)
     pb_val = row_filter._to_pb()
-    expected_pb = _RowFilterPB(value_regex_filter=regex)
+    expected_pb = _RowFilterPB(value_regex_filter=expected_byte_string)
     assert pb_val == expected_pb
 
 
 def test_exact_value_filter_to_dict_w_int():
-    import struct
-    from google.cloud.bigtable.row_filters import ExactValueFilter
+    from google.cloud.bigtable.row_filters import LiteralValueFilter
     from google.cloud.bigtable_v2.types import data as data_v2_pb2
 
     value = 1
-    regex = struct.Struct(">q").pack(value)
-    row_filter = ExactValueFilter(value)
-    expected_dict = {"value_regex_filter": regex}
+    # encode numbers with double escapes for the backslashes
+    expected_byte_string = b"\\\x00\\\x00\\\x00\\\x00\\\x00\\\x00\\\x00\\\x01"
```

Comment:
Is this the proper encoding for ints?

Reply:
I found and fixed [a bug in my null-byte detection](https://github.com/googleapis/python-bigtable/blob/804cc143792823977767604580169847cbfd5aff/google/cloud/bigtable/row_filters.py#L529), which should line up with your suggestion 1. That said, the [comments in the source snippet](https://github.com/google/re2/blob/70f66454c255080a54a8da806c52d1f618707f8a/re2/re2.cc#L479) say that it's unneeded with RE2, and my system tests seem to work with or without the special null case in place, for what it's worth.

Was that null byte the only issue you're pointing out, or are there other problems? I'm having trouble finding good test cases/documentation around RE2 parsing. [The snippet this was adapted from](https://github.com/google/re2/blob/70f66454c255080a54a8da806c52d1f618707f8a/re2/re2.cc#L456) seems to imply that all non-alphanum bytes can be escaped with a `\`, but I'm not seeing that backed up anywhere [in the docs](https://github.com/google/re2/wiki/Syntax).

Reply:
ok, I'll trust you on this one :)